CMakeLists.txt

```diff
@@ -67,8 +67,15 @@ include(cmake/SetupPackages.cmake)
 include(cmake/SetupRajaConfig.cmake)
 
 # Configure a header file with all the variables we found.
+get_directory_property( RAJA_DEFINITIONS_INITIAL DIRECTORY ${CMAKE_SOURCE_DIR} COMPILE_DEFINITIONS )
+set(RAJA_DEFINITIONS "")
+foreach(definition ${RAJA_DEFINITIONS_INITIAL})
+  set(RAJA_DEFINITIONS "${RAJA_DEFINITIONS} -D${definition}")
+endforeach()
```

Comment:
This is for anything passed in to the program with -D, which is to say anything set in a CMakeLists.txt with add_definitions. If you add something with add_definitions, it doesn't (seem to) get added to CXX_COMPILE_FLAGS (somehow?), this just also handles those